Jenkinsfile.update

```diff
@@ -0,0 +1,237 @@
+                    maven.mvnVersionsUpdateParentAndChildModules(getOptawebEmployeeRosteringMavenCommand(), getProjectVersion(), true)
+                }
+            }
+        }
+        stage('Update version.org.optaplanner') {
+            when {
+                expression { return getProjectVersion() != '' }
+            }
+            steps{
+                updateOptaplannerSnapshot('version.org.optaplanner', getProjectVersion(), kogitoExamplesRepository)
+                updateOptaplannerSnapshot('version.org.optaplanner', getProjectVersion(), kogitoAppsRepository)
+                updateQuickstartsSnapshot(getProjectVersion(), quickstartsRepository)
+            }
+        }
+        stage('Create PRs'){
+            steps {
+                commitAndCreatePR(optaplannerRepository)
+                commitAndCreatePR_II(quickstartsRepository)
+                commitAndCreatePRIgnoringNpmRegistry(vehicleRoutingRepository)
+                commitAndCreatePRIgnoringNpmRegistry(employeeRosteringRepository)
+                commitAndCreatePR(kogitoAppsRepository)
+                commitAndCreatePR(kogitoExamplesRepository)
+            }
+        }
+    }
+}
+
+// ****** GETTERS AND SETTERS PARAMS/PROPERTIES ******
+
+String getProjectVersion(){
+    return params.NEW_SNAPSHOT_VERSION
+}
+String getGitAuthor() {
+    return params.GIT_AUTHOR
+}
+String getKogitoVersion() {
+    return params.KOGITO_VERSION
+}
+String getBuildBranch() {
+    return params.BUILD_BRANCH_NAME
+}
+String getQuickstartsBuildBranch() {
+    return params.BUILD_BRANCH_NAME_QUICKSTARTS
+}
+boolean isRelease() {
+    return params.RELEASE
+}
+String getBotAuthorCredsID(){
+    return params.BOT_CREDENTIALS_ID
+}
+String getBotBranch(){
+    return "${getProjectVersion()}-${env.BOT_BRANCH_HASH}"
+}
+
+// ******  CHECKOUT REPOSITORIES ******
+void checkoutRepo(String repo, String branch, String dirName=repo) {
+    dir(dirName) {
+        deleteDir()
+        checkout(githubscm.resolveRepository(repo, getGitAuthor(), branch, false))
+    }
+}
+// ****** UPDATE VERSIONS ******
+/**
+ * Builds the parent modules and the BOM so that project depending on these artifacts can resolve.
+ */
+void mavenCleanInstallOptaPlannerParents() {
+    getOptaplannerMavenCommand()
+        .skipTests(true)
+        .withOptions(['-U', '-pl org.optaplanner:optaplanner-build-parent,org.optaplanner:optaplanner-bom', '-am'])
+        .run('clean install')
+}
+void updateOptaplannerSnapshot(String optaString, String optaVersion, String repo, String dirName=repo) {
+    dir(dirName) {
+        maven.mvnSetVersionProperty(optaString, optaVersion)
+    }
+}
+
+void updateQuickstartsSnapshot(String quickstartVersion, String repo, String dirName=repo) {
+    dir(dirName) {
+        sh "sed -i -E 's/def optaplannerVersion = \"[^\"\\s]+\"/def optaplannerVersion = \"${quickstartVersion}\"/' */build.gradle"
+        sh "git add */build.gradle"
```

Comment:
Suggesting to move git add to the place with the other files staging
githubscm.commitChanges(commitMsg, { githubscm.findAndStageNotIgnoredFiles('pom.xml') githubscm.findAndStageNotIgnoredFiles('build.gradle') })

- We will keep it in one place
- We will avoid pushing ignored files if build.gradle once will be moved to git ignore